src/components/comments/html-pin-adapter/index.ts

```diff
@@ -0,0 +1,893 @@
+
+  // Data about the current state of the application
+  private selectedPin: HTMLElement | null = null;
+  private dataAttribute: string = 'data-superviz-id';
+
+  // Coordinates/Positions
+  private mouseDownCoordinates: Simple2DPoint;
+  private commentsSide: HorizontalSide = 'left';
+  private temporaryPinCoordinates: TemporaryPinData = {};
+
+  // Elements
+  private container: HTMLElement;
+  private pinsContainer: HTMLDivElement;
+  private elementsWithDataId: Record<string, HTMLElement> = {};
+  private divWrappers: Map<string, HTMLElement> = new Map();
+  private pins: Map<string, HTMLElement>;
+  private traversedDom: Set<HTMLElement> = new Set();
+
+  // Observers
+  private resizeObserver: ResizeObserver;
+  private mutationObserver: MutationObserver;
+
+  constructor(containerId: string, dataAttributeName?: string) {
+    this.logger = new Logger('@superviz/sdk/comments-component/container-pin-adapter');
+    this.container = document.getElementById(containerId) as HTMLElement;
+
+    if (!this.container) {
+      const message = `Element with id ${containerId} not found`;
+      this.logger.log(message);
+      throw new Error(message);
+    }
+
+    this.createPinsContainer();
+    this.dataAttribute = dataAttributeName || this.dataAttribute;
+    this.isActive = false;
+    this.prepareElements();
+
+    this.mutationObserver = new MutationObserver(this.handleMutationObserverChanges);
+    this.resizeObserver = new ResizeObserver(this.handleResizeObserverChanges);
+    this.observeContainer();
+    this.observeElements();
+
+    this.pins = new Map();
+    this.onPinFixedObserver = new Observer({ logger: this.logger });
+    this.annotations = [];
+
+    document.body.addEventListener('select-annotation', this.annotationSelected);
+    Object.values(this.elementsWithDataId).forEach((element) => {
+      this.addScrollListeners(element);
+    });
+  }
+
+  // ------- setup -------
+  /**
+   * @function destroy
+   * @description destroys the pin adapter.
+   * @returns {void}
+   * */
+  public destroy(): void {
+    this.logger.log('Destroying HTML Pin Adapter for Comments');
+    this.removeListeners();
+    this.removeObservers();
+    this.divWrappers.forEach((divWrapper) => divWrapper.remove());
+    this.divWrappers.clear();
+    this.pins.forEach((pin) => pin.remove());
+    this.pins.clear();
+    delete this.divWrappers;
+    delete this.pins;
+    delete this.elementsWithDataId;
+    delete this.logger;
+    this.onPinFixedObserver.destroy();
+    delete this.onPinFixedObserver;
+    this.pinsContainer.remove();
+    delete this.pinsContainer;
+    delete this.container;
+
+    this.annotations = [];
+
+    document.body.removeEventListener('select-annotation', this.annotationSelected);
+    document.body.removeEventListener('toggle-annotation-sidebar', this.onToggleAnnotationSidebar);
+    this.traversedDom.forEach((element) => {
+      element.removeEventListener('scroll', this.onScroll);
+    });
+  }
```

Comment:
```suggestion
  public destroy(): void {
    this.logger.log('Destroying HTML Pin Adapter for Comments');
    this.removeListeners();
    this.removeObservers();
    this.divWrappers.forEach((divWrapper) => divWrapper.remove());
    this.divWrappers.clear();
    this.pins.forEach((pin) => pin.remove());
    this.pins.clear();
    this.divWrappers = undefined;
    this.pins = undefined;
    this.elementsWithDataId = undefined;
    this.logger = undefined;
    this.onPinFixedObserver.destroy();
    this.onPinFixedObserver = undefined;
    this.pinsContainer.remove();
    this.pinsContainer = undefined;
    this.container = undefined;

    this.annotations = [];

    document.body.removeEventListener('select-annotation', this.annotationSelected);
    document.body.removeEventListener('toggle-annotation-sidebar', this.onToggleAnnotationSidebar);
    this.traversedDom.forEach((element) => {
      element.removeEventListener('scroll', this.onScroll);
    });
    Object.values(this.elementsWithDataId).forEach((element) => {
      element.removeEventListener('scroll', this.onScroll)
    });
  }
```

1. Delete keyword has performance problems, it's better to set the variables as undefined. 
2. You need to remove event listeners on elements with data-attribute